concepts/regular-expressions/about.md

```diff
@@ -0,0 +1,160 @@
+# About
+
+## 1. Regular Expressions in JavaScript
+
+A [Regular Expression][mdn-regular-expressions] (or Regex) is a sequence of characters that we can use to target and manipulate certain elements in strings. Hence, we can:
+
+- **Search** for a text in a string
+- **Replace** substrings in a string
+- **Extract** information from a string
+
+> 💡 JavaScript's regex flavor is part of the ECMA-262 standard for the language. This means that you don't have to worry about browser-specific compatibility.
+
+## 2. How to create Regular Expressions
+
+1. **Regular Expression Literal**:
+
+```javascript
+const regex = /[a-z]/;
+```
+
+2. Constructor **RegExp**:
+
+```javascript
+const regex = new RegExp('[a-z]');
+```
+
+## 3. 🏴‍☠️ Flags
+
+Some of the widely used are:
+
+- `/g` - Global Search
+- `/i` - Case Insensitive
+- `/m` - Multiline Search
+
+Here is a simple example:
+
+```javascript
+const re = /home/gi;
+const str = 'Home, sweet home.';
+const myArray = str.match(re);
+console.log(myArray);
+
+=> // ["Home", "home"]
+```
+
+```javascript
+let regex = /[a-z]/gi; // literal notation
+let regex = new RegExp('[a-z]', 'gi'); // constructor with string pattern as first argument
+let regex = new RegExp(/[a-z]/, 'gi'); // constructor with regular expression literal as first argument (Starting with ECMAScript 6)
+```
+
+## 4. Most common Functions
+
+When regular expressions are combined with current build-in functions in JavaScript, we have some powerful ways of manipulating and getting data from strings.
+
+These are some of the most common functions used alongside regex.
+
+### Test
+
+```javascript
+const str = 'It is difficult to test if you have a virus';
+const result = /virus$/.test(str);
+
+console.log(result); // true
+```
+
+### Match
+
+In this way, we are able both to **search** and to **extract** information from any string. For example:
+
+```javascript
+const funnyQuote =
+  'If you see someone crying, ask if it is because of their haircut.';
+const regex1 = /someone/;
+const regex2 = /happy/;
+
+funnyQuote.match(regex1); // ["someone", index: 3, input: "If you see someone crying, ask if it is because of their haircut.", groups: undefined]
+
+funnyQuote.match(regex2); // null
+```
+
+When the Global Search flag `/g` is present, instead of getting the only match alongside useful information such as the index or input, the method returns all of the occurances displayed in the array:
```

Comment:
```suggestion
When the Global Search flag `/g` is present, instead of getting the only match alongside useful information such as the index or input, the method returns all of the occurrences present in the array:
```